pkg/sdk/testint/authentication_policies_gen_integration_test.go

```diff
@@ -0,0 +1,330 @@
+package testint
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/internal/collections"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInt_AuthenticationPolicies(t *testing.T) {
+	client := testClient(t)
+	ctx := testContext(t)
+	cert := random.GenerateX509(t)
+
+	assertAuthenticationPolicy := func(t *testing.T, authenticationPolicy *sdk.AuthenticationPolicy, id sdk.SchemaObjectIdentifier, expectedComment string) {
+		t.Helper()
+		assert.NotEmpty(t, authenticationPolicy.CreatedOn)
+		assert.Equal(t, id.Name(), authenticationPolicy.Name)
+		assert.Equal(t, id.SchemaName(), authenticationPolicy.SchemaName)
+		assert.Equal(t, id.DatabaseName(), authenticationPolicy.DatabaseName)
+		assert.Equal(t, "", authenticationPolicy.Options)
+		assert.Equal(t, "ACCOUNTADMIN", authenticationPolicy.Owner)
+		assert.Equal(t, expectedComment, authenticationPolicy.Comment)
+		assert.Equal(t, "ROLE", authenticationPolicy.OwnerRoleType)
+	}
+
+	cleanupAuthenticationPolicyProvider := func(id sdk.SchemaObjectIdentifier) func() {
+		return func() {
+			err := client.AuthenticationPolicies.Drop(ctx, sdk.NewDropAuthenticationPolicyRequest(id).WithIfExists(true))
+			require.NoError(t, err)
+		}
+	}
+
+	cleanupSecurityIntegration := func(t *testing.T, id sdk.AccountObjectIdentifier) {
+		t.Helper()
+		t.Cleanup(func() {
+			err := client.SecurityIntegrations.Drop(ctx, sdk.NewDropSecurityIntegrationRequest(id).WithIfExists(true))
+			assert.NoError(t, err)
+		})
+	}
+
+	createAuthenticationPolicy := func(t *testing.T) *sdk.AuthenticationPolicy {
+		t.Helper()
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+
+		err := client.AuthenticationPolicies.Create(ctx, sdk.NewCreateAuthenticationPolicyRequest(id))
+		require.NoError(t, err)
+		t.Cleanup(cleanupAuthenticationPolicyProvider(id))
+
+		authenticationPolicy, err := client.AuthenticationPolicies.ShowByID(ctx, id)
+		require.NoError(t, err)
+
+		return authenticationPolicy
+	}
+
+	defaultCreateRequest := func() *sdk.CreateAuthenticationPolicyRequest {
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+		comment := "some_comment"
+		return sdk.NewCreateAuthenticationPolicyRequest(id).
+			WithOrReplace(true).
+			WithComment(comment)
+	}
+
+	createSAML2Integration := func(t *testing.T, with func(*sdk.CreateSaml2SecurityIntegrationRequest)) sdk.AccountObjectIdentifier {
```

Comment:
you can use `testClientHelper().SecurityIntegration.CreateSaml2` instead

Reply:
Addressed in https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3068